js/main.js

```diff
@@ -40,6 +39,20 @@ document.addEventListener('DOMContentLoaded', function (e) {
   }
 });
 
+document.addEventListener('click', function (e) {
+  if (!e.target.hasAttribute('href')) {
+    return;
+  }
+
+  var editProfileForm = document.querySelector('main > div[data-view]');
+  if ($form.elements.username.value.trim() === '' && editProfileForm.className === 'show') {
+    return;
+  }
+
+  viewSwapping(e.target.getAttribute('data-view'));
```

Comment:
This should be in an `else` statement for the above `if` statement. We only want to allow the user to navigate to the view of their choice if they have user profile data saved already